be/src/vec/functions/function_timestamp.cpp

```diff
@@ -1182,6 +1184,139 @@ class FunctionOtherTypesToDateType : public IFunction {
+                    null_map->get_data().data()[i] = true;
+                }
+            } else if (
+                    std::regex_match(
+                            iso_date, match,
+                            year_month_regex)) { // Year and Month format (YYYY-MM, but NOT YYYYMM)
+                int year = std::stoi(match[1]);
+                int month = std::stoi(match[2]);
+
+                if (!(ts_value.template set_time_unit<YEAR>(year) &&
+                      ts_value.template set_time_unit<MONTH>(month))) {
+                    null_map->get_data().data()[i] = true;
+                }
+                ts_value.template unchecked_set_time_unit<DAY>(1);
+            } else if (std::regex_match(iso_date, match,
+                                        year_only_regex)) { // Year only format (YYYY)
+                int year = std::stoi(match[1]);
+
+                if (!ts_value.template set_time_unit<YEAR>(year)) {
+                    null_map->get_data().data()[i] = true;
+                }
+                ts_value.template unchecked_set_time_unit<MONTH>(1);
+                ts_value.template unchecked_set_time_unit<DAY>(1);
+
+            } else if (
+                    std::regex_match(
+                            iso_date, match,
+                            week_regex)) { // Week date format (YYYY-Www, YYYY-Www-D, YYYYWww or YYYYWwwD)
+                int year = std::stoi(match[1]);
+                int week = std::stoi(match[3]);
+                int weekday = match[5].matched ? std::stoi(match[5])
+                                               : 1; // Default to Monday if no day is specified
+
+                // weekday [1,7]    week [1,53]
+                if (weekday < 1 || weekday > 7 || week < 1 || week > 53) {
+                    null_map->get_data().data()[i] = true;
+                    continue;
+                }
+
+                auto first_day_of_week = getFirstDayOfISOWeek(year);
+                ts_value.template unchecked_set_time_unit<YEAR>(
+                        first_day_of_week.year().operator int());
+                ts_value.template unchecked_set_time_unit<MONTH>(
+                        first_day_of_week.month().operator unsigned int());
+                ts_value.template unchecked_set_time_unit<DAY>(
+                        first_day_of_week.day().operator unsigned int());
+
+                auto day_diff = (week - 1) * 7 + weekday - 1;
+                TimeInterval interval(DAY, day_diff, false);
+                ts_value.date_add_interval<DAY>(interval);
+
+            } else if (std::regex_match(
+                               iso_date, match,
+                               day_of_year_regex)) { // Day of year format (YYYYDDD or YYYY-DDD)
+                int year = std::stoi(match[1]);
+                int day_of_year = std::stoi(match[3]);
+
+                if (is_leap(year)) {
+                    if (day_of_year < 0 || day_of_year > 366) {
```

Comment:
better for judgement like this, set it to `[[unlikely]]`

Reply:
i will  add it